src/Exceptions/InvalidFilterQuery.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Spatie\QueryBuilder\Exceptions;
+
+use Illuminate\Http\Response;
+use Illuminate\Support\Collection;
+
+class InvalidFilterQuery extends InvalidQuery
+{
+    /** @var \Illuminate\Support\Collection */
+    public $unknownFilters;
+
+    /** @var \Illuminate\Support\Collection */
+    public $allowedFilters;
+
+    public function __construct(Collection $unknownFilters, Collection $allowedFilters)
+    {
+        $this->unknownFilters = $unknownFilters;
+        $this->allowedFilters = $allowedFilters;
+
+        $unknownFilters = $this->unknownFilters->implode(', ');
+        $allowedFilters = $this->allowedFilters->implode(', ');
+        $message = "Given filter(s) `{$unknownFilters}` are not allowed. Allowed filters are `{$allowedFilters}`.";
```

Comment:
Use `filter(s)` instead of `filters` to be more consistent. Do it also for the other exceptions.